### DISCUSSION TITLE
Is it  possible to implement test time adaptation using avalanche?

### DISCUSSION THREAD

**Discussion by Amos58** [ContinualAI/avalanche#1616] [Feature Request]
Continual learning is implemented by a continuous train-test paradigm, but this can not satisfy my needs that I need to tune my network parameters only during test phase. Test time adaptation satifies my nedds and it is similar to continual learning, I wonder if it is possible to implement test time adaptation using avalanche. Thank you.

**Comment by AntonioCarta** [comment]
We don't have TTA methods, but Avalanche strategies provide callbacks that you can use to inject code at any point of the train/eval loops. You can use those to add TTA methods to Avalanche strategies.

**Reply by Amos58** [reply]
thank you, I'll look at the API documentation and look into how to implement it
